hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java

```diff
@@ -205,4 +221,38 @@ public void testUpdateQueryCausesInvalidationOfEntireRegion() {
         assertEquals(10, dummyEntityCacheStats.getMissCount());
         assertEquals(0, dummyEntityCacheStats.getHitCount());
     }
+
+    @Test
+    public void testNaturalIdCacheStillHitsAfterIrrelevantNaturalIdUpdate() {
```

Comment:
i split the test into two. one checks if natural id eviction works on updates. the other ensures the eviction is done for only affected entities, not the whole table